src/app/js/formats/distributionChart/global-radar-chart/RadarChartAdmin.js

```diff
@@ -75,10 +76,86 @@ class RadarChartAdmin extends Component {
         updateAdminArgs('scale', scale, this.props);
     };
 
+    setColors = (_, colors) => {
+        updateAdminArgs('colors', colors, this.props);
+    };
+
+    setColorsWithColorPicker = () => {
+        updateAdminArgs('colors', this.getColorsFromPicker(), this.props);
+    };
+
+    getColorsFromPicker() {
+        let numberOfPickers = this.getColorsArray().length;
+        let res = '';
+        for (var i = 0; i < numberOfPickers; i++) {
+            if (i == 0) {
+                res += this.state.colors[i].color;
+            } else {
+                res += ' ' + this.state.colors[i].color;
+            }
+        }
+        this.setColors;
+        return res;
+    }
+
+    getColorsArray() {
+        return this.props.args.colors.split(' ');
+    }
+
+    constructor(props) {
+        super(props);
+        isFirstLoading = true; // to initialize values on admin UI opening
+        const colorsArray = this.getColorsArray();
+
+        this.state = {
+            colors: [{ color: colorsArray[0] }],
+        };
+
+        let numberOfPickers = this.getColorsArray().length;
+        for (let index = 1; index < numberOfPickers; index++) {
+            this.state = {
+                colors: [...this.state.colors, { color: colorsArray[index] }],
+            };
+        }
```

Comment:
I would have used:

```js
        this.state = {
            colors: colorsArray.map(color => ({ color }))
        };
```

Reply:
Oh I was stuck for a long time with this assignment (the only way I have found was the 0 + loop), but this mapping works fine ! Thanks !